pubber/src/main/java/daq/pubber/Pubber.java

```diff
@@ -858,19 +858,22 @@ private void configHandler(Config config) {
   }
 
   private void processConfigUpdate(Config config) {
-    final int actualInterval;
+    int actualInterval;
     if (config != null) {
       deviceConfig = config;
       info(String.format("%s received config %s", getTimestamp(), isoConvert(config.timestamp)));
       deviceState.system.last_config = config.timestamp;
-      actualInterval = updateSystemConfig(config.pointset);
-      updatePointsetConfig(config.pointset);
+      actualInterval = updatePointsetConfig(config.pointset);
+      updatePointsetPointsConfig(config.pointset);
       updateDiscoveryConfig(config.discovery);
       extractEndpointBlobConfig();
     } else {
       info(getTimestamp() + " defaulting empty config");
       actualInterval = DEFAULT_REPORT_SEC * 1000;
     }
+    if (configuration.options.fixedSampleRate != null) {
+      actualInterval = configuration.options.fixedSampleRate * 1000;
```

Comment:
a better coding style is to avoid mutable variables, and instead intoruce a new variable with the new maning, eith.

`useInterval = configuration.options.fixedSampleRate == null ? actualInterval : configuration.options.fixedSampleRate * 1000`

In the big picture it makes the code more readable because you don't need to keep track of when a variable might change. Mutable variables are only really necessary in loops, etc... (and this is why the variable was initially indicated as final, which forces that it's written ones, and exactly once)

Reply:
Thanks for the feedback, changed as recommended